minari/dataset/minari_storage.py

```diff
@@ -87,17 +88,53 @@ def apply(
 
         return out
 
+    def _h5_group_to_dict_recursive(self, hdf_ref, timestep):
+
+        if not isinstance(hdf_ref, h5py.Group):
+            return hdf_ref[timestep]
+
+        if "tuple" in hdf_ref.attrs.keys():
```

Comment:
is the attribute tuple needed? Can we get if it a tuple from the space?

Reply:
It could be obtained from the space but it would be a bit awkward since we are recursing on the nested h5py structure. We could also mirror the recursion on the space as well as the h5py group if we want to avoid saving the `tuple` attributes. Let me know which way you'd prefer.

Reply:
I added mirroring on the space to check for Tuple spaces, and also removed the tuple attribute saving when saving to hdf5 files.